server/sockets/cohort.js

```diff
@@ -134,52 +136,52 @@ const updateCohort = (socket, allCohortsViewClients) => {
 const updateCohortHeaderStatus = (socket, cohortClientLocks) => {
   socket.on(CohortHeaderStatusTypes.UNLOCK, data => {
     const { cohortId, descriptionLock, nameLock, userId } = data;
+    const locks = { description: descriptionLock, name: nameLock };
 
-    socket.broadcast
-      .to(cohortChannel(cohortId))
-      .emit(CohortHeaderStatusTypes.UNLOCK, data);
+    handleLocks(Cohort, { _id: cohortId, ownerIds: userId })(locks).then(() => {
+      socket.broadcast
+        .to(cohortChannel(cohortId))
+        .emit(CohortHeaderStatusTypes.UNLOCK, { cohortId, locks });
 
-    if (cohortClientLocks.has(userId)) {
-      clearTimeout(cohortClientLocks.get(userId));
-      cohortClientLocks.delete(userId);
-    }
+      const lock = isDefined(nameLock) ? lockName : lockDescription;
 
-    const locks = { description: descriptionLock, name: nameLock };
-
-    handleLocks(Cohort, { _id: cohortId })(locks);
+      if (cohortClientLocks.has(lock(cohortId))) {
+        clearTimeout(cohortClientLocks.get(lock(cohortId)));
+        cohortClientLocks.delete(lock(cohortId));
+      }
+    });
   });
 
   socket.on(CohortHeaderStatusTypes.LOCK, data => {
     const { cohortId, descriptionLock, nameLock, userId } = data;
+    const locks = { description: descriptionLock, name: nameLock };
 
-    socket.broadcast
-      .to(cohortChannel(cohortId))
-      .emit(CohortHeaderStatusTypes.LOCK, data);
-
-    const delayedUnlock = setTimeout(() => {
-      const { cohortId } = data;
-      const updatedData = { cohortId };
-
-      if (isDefined(nameLock)) {
-        updatedData.nameLock = false;
-      }
-
-      if (isDefined(descriptionLock)) {
-        updatedData.descriptionLock = false;
-      }
-
+    handleLocks(Cohort, { _id: cohortId, ownerIds: userId })(locks).then(() => {
       socket.broadcast
-        .to(`cohort-${cohortId}`)
-        .emit(CohortHeaderStatusTypes.UNLOCK, updatedData);
-
-      cohortClientLocks.delete(userId);
-    }, LOCK_TIMEOUT);
-
-    cohortClientLocks.set(userId, delayedUnlock);
+        .to(cohortChannel(cohortId))
+        .emit(CohortHeaderStatusTypes.LOCK, { cohortId, locks });
+
+      const lock = isDefined(nameLock) ? lockName : lockDescription;
+      const delayedUnlock = setTimeout(() => {
+        if (isDefined(nameLock)) {
+          locks.name = false;
+        } else {
```

Comment:
unnecessary else

Reply:
it is necessary, without else description lock will always set to false. In lists and cohort case one user can set name OR description lock so I had to use else. I can use:
```
if (isDefined(nameLock)) {
          locks.name = false;
        } 
if (isDefined(descriptionLock)) {
          locks.description = false;
        } 
```
but in this case we would have unnecessary check

Reply:
I think this is better and more reasonable:
```
if (isDefined(nameLock)) {
          locks.name = false;
        } 
if (isDefined(descriptionLock)) {
          locks.description = false;
        } 
```